src/components/Swap/SwapActionButton/useOnSubmit.ts

```diff
@@ -6,16 +6,20 @@ import { displayTxHashAtom, Field } from 'state/swap'
 import { TransactionInfo, TransactionType } from 'state/transactions'
 import { isAnimating } from 'utils/animations'
 
-/** Submits a transaction. */
+/**
+ * Returns a callback to submit a transaction.
+ *
+ * Returns a boolean indicating whether the transaction was submitted.
```

Comment:
Could you also specify that this will throw (and not just return false) if something besides a user rejection occurs?